.github/workflows/integration-test-review.yml

```diff
@@ -32,21 +32,19 @@ jobs:
         uses: actions/github-script@v7
         with:
           script: |
-            const { number, html_url } = (await github.rest.repos.listPullRequestsAssociatedWithCommit({
-              commit_sha: context.sha,
-              owner: "${{ github.event.workflow_run.head_repository.owner.login }}",
-              repo: "${{ github.event.workflow_run.head_repository.name }}",
-            })).data[0] ?? {};
+            const { html_url } = await github.rest.issues.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+            });
```

Comment:
This intuitively makes more sense; I was struggling to understand why `listPullRequestsAssociatedWithCommit` was always returning the same issue, but perhaps it has to do with the tests running on a merge commit, which would possibly bring in all PRs branched from `main`? :confused: I'm especially struggling with why all the comments were going to the same PR, which would periodically change. This is much more intuitive.

Reply:
(Finally circling around to look at this.) 

I'm honestly not sure either. We're using a very similar GraphQL query on pushes to a branch to get the PR associated with a merge commit and the labels their within, which has not had these problems:
https://github.com/ASFHyP3/actions/blob/develop/.github/workflows/reusable-bump-version.yml#L25-L49

I don't see much difference between the queries other than we specify the sorting:
https://github.com/ASFHyP3/actions/blob/develop/.github/workflows/reusable-bump-version.yml#L25-L49